sql-plugin/src/main/java/com/nvidia/spark/rapids/GpuCompressedColumnVector.java

```diff
@@ -33,9 +33,19 @@
   private final DeviceMemoryBuffer buffer;
   private final TableMeta tableMeta;
 
+  /**
+   * Build a columnar batch from a compressed table.
+   * NOTE: The data remains compressed and cannot be accessed directly from the columnar batch.
+   */
   public static ColumnarBatch from(CompressedTable compressedTable) {
-    DeviceMemoryBuffer buffer = compressedTable.buffer();
-    TableMeta tableMeta = compressedTable.meta();
+    return from(compressedTable.buffer(), compressedTable.meta());
+  }
+
+  /**
+   * Build a columnar batch from a compressed data buffer and specified table metadata
+   * NOTE: The data remains compressed and cannot be accessed directly from the columnar batch.
+   */
+  public static ColumnarBatch from(DeviceMemoryBuffer buffer, TableMeta tableMeta) {
```

Comment:
Just a nit:
 
```suggestion
  public static ColumnarBatch from(DeviceMemoryBuffer compressedBuffer, TableMeta tableMeta) {
```

Reply:
So you can't really apply the suggestion on its own, but the idea is just to make it clear(er) that this is a compressed buffer.